azerothcore/azerothcore-wotlk
# feat(Core/Player): Additional option for worldserver config parameter "InstantFlightPaths"

##### CHANGES PROPOSED:
Add an additional option, so the worldserver config parameter "InstantFlightPaths" now takes 3 values:
- 0: Instant flight permanently off (default)
- 1: Instant flight permanently on
- 2 (new value): Instant flight is on, but the player may toggle it between off and on at each flight master via a gossip option

Some of the flight paths show a very nice scenery, so the additional option allows the player to choose if he wants instant flight or not.

##### ISSUES ADDRESSED:
none

##### TESTS PERFORMED:
- tested build on Ubuntu 16.04 / clang 7
- tested successfully in-game

##### HOW TO TEST THE CHANGES:
Use different values for "InstantFlightPaths" and check the flight paths at any flight master:

- 0: Instant flight is off and the player does not get the gossip option to toggle instant flight
- 1: Instant flight is on and the player does not get the gossip option to toggle instant flight
- 2: The player gets an additional gossip option to toggle instant flight

Please also try flight paths triggered by spells, those should never be instant, no matter the configuration (e.g. Surristrasz (24795) which provides a flight when choosing "I'd like passage to the Transitus Shield).

##### KNOWN ISSUES AND TODO LIST:
none

##### Target branch(es):
Master
 
## How to test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here in the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR
